.github/workflows/continuous-integration.yml

```diff
@@ -243,11 +243,81 @@ jobs:
         run: yarn build --verbose --buildtype=${{ matrix.buildtype }}
         timeout-minutes: 30
 
+      - name: Generate build details
+        run: |
+          cat > build/${{ matrix.buildtype }}/BUILD.txt << EOF
+          BUILDTYPE=${{ matrix.buildtype }}
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          RUN_ID=${{ github.run_id }}
+          REF=${{ github.sha }}
+          BUILDTIME=$(date +%s)
+          EOF
```

Comment:
Moved the generation of BUILD.txt into the build step. That way, the archive step won't have to do an extra set of decompression and compression.

Even if the step were to only add the file to the existing tar file, I figured having the artifacts be consistent with the S3 uploaded files (by having both with the BUILD.txt) would be better and maybe reduce confusion down the line.

Also felt right to create `BUILD.txt` at the time of the actual build.